src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java

```diff
@@ -221,8 +222,8 @@ public void overrideMaxDisplaySize(Long maxDisplaySize) {
 
     private void addCommonFilterTemplates(List<String> filters) {
 
-        if (isNotBlank(accountId)) {
-            filters.add(" t.gateway_account_id = :" + GATEWAY_ACCOUNT_EXTERNAL_FIELD);
+        if (!isEmpty(accountIds)) {
```

Comment:
I still think we can just do `accountIds.isEmpty()`: `setAccountIds(…)` will throw a NPE if `accountIds` is null (because of the `List.copyOf(…)`) and the only caller of that method is in `TransactionResource`’s `streamCsv(…)` method, which will throw its own NPE if the parameter isn’t there. So by this point I’m fairly certain we can be sure that `accountIds` isn’t null.

Reply:
Whoops, I was wrong here! It’s possible `setAccountIds(…)` is never called at all, so `accountIds` remains at its initial null value.